sites/shared/components/workbench/inputs/measurement.mjs

```diff
@@ -11,16 +11,18 @@ import { measurementAsMm } from 'shared/utils.mjs'
  * m holds the measurement name. It's just so long to type
  * measurement and I always have some typo in it because dyslexia.
  */
-export const MeasurementInput = ({ m, gist, app, updateMeasurements, focus }) => {
+export const MeasurementInput = ({ m, gist, app, updateMeasurements, focus, optional }) => {
   const { t } = useTranslation(['app', 'measurements'])
   const prefix = app.site === 'org' ? '' : 'https://freesewing.org'
   const title = t(`measurements:${m}`)
 
   const isDegree = isDegreeMeasurement(m)
   const factor = useMemo(() => (isDegree ? 1 : gist.units == 'imperial' ? 25.4 : 10), [gist.units])
 
-  const isValValid = (val) =>
-    typeof val === 'undefined' || val === '' ? null : val != false && !isNaN(val)
+  const isValValid = (val) => {
+    if (optional && val === 0) return true
+    return typeof val === 'undefined' || val === '' ? null : val != false && !isNaN(val)
```

Comment:
I would just replace these lines with `return val  == false ? optional : !isNaN(val)` because `undefined`, `0`, and `''` are all falsy, and any of them should be allowed on an optional measurement

Reply:
Replaced code as suggested.